posthog/async_migrations/migrations/0001_events_sample_by.py

```diff
@@ -0,0 +1,133 @@
+from constance import config
+
+from ee.clickhouse.client import sync_execute
+from ee.clickhouse.sql.events import EVENTS_TABLE, EVENTS_TABLE_SQL
+from posthog.async_migrations.definition import AsyncMigrationDefinition, AsyncMigrationOperation
+from posthog.constants import AnalyticsDBMS
+from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_DATABASE
+from posthog.version_requirement import ServiceVersionRequirement
+
+TEMPORARY_TABLE_NAME = "temp_events_0001_events_sample_by"
+
+
+class Migration(AsyncMigrationDefinition):
+
+    description = "Events table migration for compatible sample by column."
+
+    posthog_min_version = "1.30.0"
+    posthog_max_version = "1.31.0"
+
+    service_version_requirements = [
+        ServiceVersionRequirement(service="clickhouse", supported_version=">=21.6.0,<21.7.0"),
+    ]
+
+    operations = [
+        AsyncMigrationOperation(
+            database=AnalyticsDBMS.CLICKHOUSE,
+            sql=EVENTS_TABLE_SQL().replace(EVENTS_TABLE, TEMPORARY_TABLE_NAME, 1),
```

Comment:
@fuziontech on your PR this had a bug where it would create a table using the current events table schema. this fixes that

Reply:
That wasn't a bug. Try running this on an instance that has materialized columns. You will be in a broken state after the migration.

Reply:
Great point. What I did see though was this:
https://github.com/PostHog/posthog/blob/7eccd3714806ac83216f14b6c125fd3ccb763b5e/posthog/special_migrations/migrations/events_sample_by_migration.py#L29

Which actually didn't do the migration. I see you updated it now. That's pretty cool syntax actually

Reply:
The other PR definitely wasn't ready to ship. WIP/Draft mode for sure.